daemon.go

```diff
@@ -146,3 +147,41 @@ func (d *Daemon) listen() {
 		go d.handleConn(c)
 	}
 }
+
+func clearUnixSockets(maddr ma.Multiaddr) error {
+	for _, subMaddr := range ma.Split(maddr) {
+		if subMaddr.Protocols()[0].Code == ma.P_UNIX {
+			socket, err := subMaddr.ValueForProtocol(ma.P_UNIX)
+			if err != nil {
+				return err
+			}
+			if err := os.Remove(socket); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (d *Daemon) Close() error {
+	if err := d.host.Close(); err != nil {
+		return err
+	}
+
+	listenMaddr := d.listener.Multiaddr()
+	if err := d.listener.Close(); err != nil {
+		return err
+	}
+
+	for _, maddr := range d.handlers {
+		if err := clearUnixSockets(maddr); err != nil {
```

Comment:
these are owned by the client, I am not sure it's appropriate to close them here.

Reply:
Agree!

Reply:
Removing the handler clearing